h/static/scripts/controllers/autosuggest-dropdown-controller.js

```diff
@@ -0,0 +1,454 @@
+      return;
+    }else if(currentActive) {
+      currentActive.classList.remove(this.options.classNames.activeItem);
+    }
+
+    if(hovering){
+      target.classList.add(this.options.classNames.activeItem);
+    }
+  }
+
+  /**
+   * _toggleSuggestionsVisibility - this function piggy backs on the setElementState
+   *  style of defining element state in its class. Used in combination with the
+   *  this.options.classNames.container the consumer has easy access to what the visibility state
+   *  of the container is.
+   *
+   * @param  {bool} show should we update the state to be visible or not
+   */
+  _toggleSuggestionsVisibility(show) {
+
+    // keeps the internal state synced with visibility
+    this.setState({
+      visible: !!show,
+    });
+  }
+
+  /**
+   * @returns {HTMLElement}  the active list item element
+   */
+  _getActiveListItemElement(){
+    return this._listContainer.querySelector('.' + this.options.classNames.activeItem);
+  }
+
+  /**
+   * _keyboardSelectionChange - navigate the list, toggling the active item,
+   *  based on the users arrow directions
+   *
+   * @param  {bool} down is the user navigating down the list?
+   */
+  _keyboardSelectionChange(down){
+
+    const currentActive = this._getActiveListItemElement();
+    let nextActive;
+
+    // we have a starting point, navigate on siblings of current
+    if(currentActive){
+      currentActive.classList.remove(this.options.classNames.activeItem);
+
+      if(down){
+        nextActive = currentActive.nextSibling;
+      }else {
+        nextActive = currentActive.previousSibling;
+      }
+
+    // we have no starting point, let's navigate based on
+    // the directional expectation of what the first item would be
+    }else if(down){
+      nextActive = this._listContainer.firstChild;
+    }else {
+      nextActive = this._listContainer.lastChild;
+    }
+
+    if(nextActive){
+      nextActive.classList.add(this.options.classNames.activeItem);
+    }
+  }
+
+  /**
+   * _renderContentContainers - build the DOM structure that makes up
+   *  the suggestion box and content containers.
+   */
+  _renderContentContainers(){
+
+    // container of all suggestion elements
+    this._suggestionContainer = document.createElement('div');
+    this._suggestionContainer.classList.add(this.options.classNames.container);
+
+
+    // child elements that will be populated by consumer
+    this._header = document.createElement('h4');
+    this._header.classList.add(this.options.classNames.header);
+    this._setHeader(this.state.header);
+    this._suggestionContainer.appendChild(this._header);
+
+    this._listContainer = document.createElement('ul');
+    this._listContainer.classList.add(this.options.classNames.list);
+    this._suggestionContainer.appendChild(this._listContainer);
+
+    // put the suggestions adjacent to the input element
+    // firefox does not support insertAdjacentElement
+    if(HTMLElement.prototype.insertAdjacentElement){
```

Comment:
I would be inclined to keep just the Firefox-safe code path here, that way you've got less to test.